agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
+	if cfg.ContainerMetadataEnabled {
+		// Get the oldest Docker Client version with up to date inspect API
+		// as some metadata is unavailable in older API client versions
+		// If this version is unavailable we use default
```

Comment:
The feature still works for older versions of docker. The differences in the versions do not actually matter previously because I get the relevant networking metadata through an alternative route for older versions. With the ENI feature added, this difference is significant because older versions of Docker API won't display the additional network modes. Again, I am not very familiar with ENI project except that I was asked to make the Network information into a list to support multiple network modes. I would think if a user wants to use this feature even with an older version of Docker, we should not disable it if it works for that older client.